test/mixed.js

```diff
@@ -173,6 +173,15 @@ describe('Mixed Types ', () => {
     );
   });
 
+  it('should limit values with a ref', async () => {
+    let context = { someValues: [1,2,3] };
+    let inst = mixed().oneOf(ref('$someValues'));
```

Comment:
this is cool but i think it behaves differently than what someone would expect. I _ think_ what folks are asking for is that the array of values the ref resolves to be the list of valid values, as in 1, 2, or 3 would be valid values. That is a good bit harder to implement and i'm not sure that it'd be worth it